### PR TITLE
[PKG-1932] laspy 2.4.1  :snowflake:

### PR DESCRIPTION
**Pull request by skupr-anaconda**
License: https://github.com/laspy/laspy/blob/2.4.1/LICENSE.txt
Requirements:
- https://github.com/laspy/laspy/blob/2.4.1/pyproject.toml
- https://github.com/laspy/laspy/blob/2.4.1/setup.py

Actions:
1. Add `--no-deps --no-build-isolation` to `script`
2. Skip `s390x`
3. Remove `requirements/build` section
4. Add `setuptools` & `wheel` to `host`, pin `lazrs-python` to `0.5.0`
5. Update `run` dependencies
6. Add `pip check`

Notes:
- it requires https://github.com/AnacondaRecipes/lazrs-python-feedstock/pull/1

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,32 +11,33 @@ source:
 
 build:
   number: 0
-  skip: true  # [py<39]
-  script: {{ PYTHON }} -m pip install . -vv
+  # maturin isn't available on s390x
+  skip: true  # [py<39 or s390x]
```
Comment (Jrice1317) [inline recipe/meta.yaml]: Maybe we should change this to `[py<37]` to correspond with upstream

https://github.com/laspy/laspy/blob/2.4.1/setup.py#L24

Comment (Jrice1317) [reply recipe/meta.yaml]:
```suggestion
  skip: true  # [py<37 or s390x]
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -11,32 +11,33 @@ source:
...
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
 
 requirements:
-  build:
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-    - numpy                                  # [build_platform != target_platform]
   host:
     - python
+    - lazrs-python 0.5.0
```
Comment (Jrice1317) [inline recipe/meta.yaml]: According to [this](https://github.com/laspy/laspy/blob/2.4.1/pyproject.toml), we only need `setuptools` and `wheel`. Can you please help me understand why we have `lazrs-python` in the `host` section as well as `run` when it seems to be an extra dependency according to [this](https://github.com/laspy/laspy/blob/2.4.1/setup.py#L38)?

Comment (Jrice1317) [reply recipe/meta.yaml]: Please correct my understanding that if `numpy` is a dependency, then it should be listed both in the `run` and `host` section just like `python`?

Comment (skupr-anaconda) [reply recipe/meta.yaml]: I added LAZ support intentionally, see https://github.com/laspy/laspy#installation to enable LAZ backend https://github.com/laspy/laspy#features. I'll make a comment about this in the recipe

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Yes, I'll change it but first I want to rebuild `lazrs-python` https://github.com/AnacondaRecipes/lazrs-python-feedstock/pull/1 where this `py<39` was introduced by mistake, IMO

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: I removed lazrs-python 0.5.0 from host

**Comment by skupr-anaconda** [comment]
Waiting https://github.com/AnacondaRecipes/lazrs-python-feedstock/pull/1

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: done

**Review by ELundby45 — APPROVED** [review]

File: abs.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+upload_channels:
```
Comment (Jrice1317) [inline abs.yaml]: Do we still need this or can it be deleted?

Comment (skupr-anaconda) [reply abs.yaml]: I removed it

**Comment by danpetry** [comment]
@Jrice1317 do you have any remaining changes required to be addressed, or are you able to give this an approval if not?

**Review by akabanovs — APPROVED** [review]
LGTM

**Pull request merged**